cmd/kubedownscaler/main.go

```
@@ -61,60 +90,18 @@ func main() {
 
 	ctx, cancel := context.WithCancel(context.Background())
 
-	downscalerNamespace, err := kubernetes.GetCurrentNamespace()
-	if err != nil {
-		slog.Warn("couldn't get namespace or running outside of cluster; skipping leader election", "error", err)
-	}
-
-	if !config.LeaderElectionEnabled {
+	if !config.LeaderElection {
 		runWithoutLeaderElection(client, ctx, &layerCli, &layerEnv, config)
 		return
 	}
 
-	runWithLeaderElection(client, downscalerNamespace, cancel, ctx, &layerCli, &layerEnv, config)
-}
-
-func initializeConfig() *util.RuntimeConfiguration {
-	config := &util.RuntimeConfiguration{
-		DryRun:                false,
-		Debug:                 false,
-		Once:                  false,
-		LeaderElectionEnabled: false,
-		Interval:              defaultInterval,
-		IncludeNamespaces:     nil,
-		IncludeResources:      []string{"deployments"},
-		ExcludeNamespaces:     util.RegexList{regexp.MustCompile("kube-system"), regexp.MustCompile("kube-downscaler")},
-		ExcludeWorkloads:      nil,
-		IncludeLabels:         nil,
-		TimeAnnotation:        "",
-		Kubeconfig:            "",
-	}
-	config.ParseConfigFlags()
-
-	return config
-}
-
-func initializeLayers() (values.Layer, values.Layer) {
-	// Initialize layers
-	layerCli := values.NewLayer()
-	layerEnv := values.NewLayer()
-
-	// Configure layerCli
-	layerCli.GracePeriod = defaultGracePeriod
-	layerCli.DownscaleReplicas = defaultDownscaleReplicas
-	layerCli.ParseLayerFlags()
-	flag.Parse()
-
-	return layerCli, layerEnv
-}
-
-func setupEnv(layerEnv *values.Layer) error {
-	err := layerEnv.GetLayerFromEnv()
+	downscalerNamespace, err := kubernetes.GetCurrentNamespace()
 	if err != nil {
-		return fmt.Errorf("failed to get layer from env: %w", err)
+		slog.Warn("couldn't get namespace or running outside of cluster; skipping leader election", "error", err)
+		runWithoutLeaderElection(client, ctx, &layerCli, &layerEnv, config)
```

```suggestion
		slog.Error("failed to get namespace or running outside of cluster; disable leader election to continue", "error", err)
		os.Exit(1)
```

---

Just to reiterate from my previous comment on this. I think it would be good to if someone explicitly enables leader election via the cli argument, we should enforce that wish. This could otherwise easily lead to unwanted behavior from wrong configuration. One of the main improvements of the gokubedownscaler is that unclear/wrong configuration will throw an error to show the user their fault, instead of just ignoring it and doing unexpected things.